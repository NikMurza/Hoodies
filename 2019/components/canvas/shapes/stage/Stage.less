.canvas-stage {
  width: 100%;
  height: 100%;
  overflow: auto;
  background-color: #f2f3f4;
}
