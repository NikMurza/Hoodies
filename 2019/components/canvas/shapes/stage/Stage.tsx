import * as React from 'react';
import './Stage.less';
import { Stage as KStage } from 'react-konva';
import Konva from 'konva';

interface IProps {
  width?: number;
  height?: number;

  onContextMenu?(evt: Konva.KonvaEventObject<PointerEvent>): void;
  onClick?(evt: Konva.KonvaEventObject<MouseEvent>): void;
}

class Stage extends React.Component<IProps> {
  public static defaultProps = {
    width: 0,
    height: 0,
  };

  protected etlRef: React.RefObject<HTMLDivElement>;

  constructor(props) {
    super(props);
    this.etlRef = React.createRef();
  }

  public componentDidMount() {
    this.setCenterScrollPosition();
  }

  public componentDidUpdate(prevProps: IProps) {
    const { width, height } = this.props;
    if ((!prevProps.width && width) || (!prevProps.height && height)) {
      this.setCenterScrollPosition();
    }
  }

  private setCenterScrollPosition() {
    if (this.etlRef.current) {
      const { scrollWidth, scrollHeight, offsetWidth, offsetHeight } = this.etlRef.current;
      this.etlRef.current.scrollTo({
        left: (scrollWidth - offsetWidth) / 2,
        top: (scrollHeight - offsetHeight) / 2,
      });
    }
  }

  public render() {
    const { width, height, onContextMenu, onClick, children } = this.props;
    return (
      <div ref={this.etlRef} className="canvas-stage">
        <KStage
          width={width}
          height={height}
          offsetX={-width / 2}
          offsetY={-height / 2}
          onContextMenu={onContextMenu}
          onClick={onClick}
        >
          {children}
        </KStage>
      </div>
    );
  }
}

export default Stage;
