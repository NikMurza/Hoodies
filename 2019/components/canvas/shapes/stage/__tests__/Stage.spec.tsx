import * as React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Stage from '../Stage';

describe('CanvasStage component', () => {
  it('render default', async () => {
    expect(toJson(shallow(<Stage />))).toMatchSnapshot();
  });
});
