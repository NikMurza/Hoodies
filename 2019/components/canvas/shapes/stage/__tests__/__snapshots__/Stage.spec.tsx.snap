// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CanvasStage component render default 1`] = `
<div
  className="canvas-stage"
>
  <ForwardRef
    height={0}
    offsetX={-0}
    offsetY={-0}
    width={0}
  />
</div>
`;
